packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -723,6 +723,10 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   /// be transparent.
   Color get barrierColor;
 
+  /// Whether this route represents an opaque dialog that covers the entire
+  /// navigator. It allows the child page to adapt its presentation accordingly.
+  bool get fullscreenDialog => false;
```

Comment:
I don't think this should be in the widgets layer. It's not used by anything else in the widgets layer.